include/zephyr/bluetooth/bluetooth.h

```diff
@@ -114,6 +114,22 @@ struct bt_le_ext_adv_cb {
 	 */
 	void (*scanned)(struct bt_le_ext_adv *adv,
 			struct bt_le_ext_adv_scanned_info *info);
+
+#if defined(CONFIG_BT_PRIVACY)
+	/**
+	 * @brief The RPA validity of the advertising set has timed out.
+	 *
+	 * This callback notifies the application that the RPA validity of
+	 * the advertising set has timed out. The user can use this callback
+	 * to synchronize the advertising payload update with the RPA rotation.
+	 *
+	 * @param adv  The advertising set object.
+	 *
+	 * @return true to rotate the current RPA, or false to use it for the
+	 *         next rotation period.
+	 */
+	bool (*rpa_timed_out)(struct bt_le_ext_adv *adv);
```

Comment:
I don't have a super strong opinion on this, but wouldn't `rpa_expired` sound better (and shorter? :)

Reply:
Well, I can do that. The ``rpa_expired`` seems simpler than the current callback name, and the expiration is frequently used for referring to timeouts.

However, the Bluetooth ``id.c`` source code uses the ``timeout`` term in the RPA context and the related HCI command is called ``LE Set Resolvable Private Address Timeout``. It is probably easier for users to associate this API with the ```bt_le_set_rpa_timeout``` API with the current name. Let me know what you think.

Alternatively, I can switch to the ``rpa_timeout`` callback name, but this one does not contain any verb and probably violates API naming rules.

Reply:
We use `timeout` for the scan "timeout", so maybe `rpa_timeout` will simply also work? But otherwise I agree with Johan; `expired` sounds better than `timed_out`.

Reply:
I change the callback name to ``rpa_expired``